pkg/test/runner/runner.go

```diff
@@ -206,7 +206,17 @@ func (r *Runner) runFnRender() error {
 	// run function
 	kptArgs := []string{"fn", "render", pkgPath}
 	for i := 0; i < r.testCase.Config.RunCount(); i++ {
-		output, fnErr := runCommand("", r.kptBin, kptArgs)
+		output, stderr, fnErr := runCommand("", r.kptBin, kptArgs)
+		if fnErr != nil {
+			if r.testCase.Config.ExitCode != 0 {
+				expected := r.testCase.Config.StdErr
+				if !strings.Contains(stderr, expected) {
+					return fmt.Errorf("wanted %s, got %s", expected, stderr)
+				}
+				return nil
+			}
+			break
```

Comment:
Don't break here. We need to compare the results even when there is an error.

Reply:
fixed!